differ/collectionsDiffer.go

```diff
@@ -13,31 +13,22 @@ func NewDefaultCollectionsDiffer() *defaultCollectionsDiffer {
 
 func (dcd *defaultCollectionsDiffer) Diff(incomingCollectionUuids []string, oldCollectionUuids []string) (map[string]bool) {
 	diffCol := make(map[string]bool)
-	for _, incColUuid := range incomingCollectionUuids {
-		found := false
-		for _, oldColUuid := range oldCollectionUuids {
-			if incColUuid == oldColUuid {
-				found = true
-				break
-			}
-		}
-		if !found {
-			diffCol[incColUuid] = false
-		}
+
+	oneWayDiff(incomingCollectionUuids, oldCollectionUuids, false, diffCol)
+	oneWayDiff(oldCollectionUuids, incomingCollectionUuids, true, diffCol)
+
+	return diffCol
+}
+
+func oneWayDiff(firstCollection []string, secondCollection []string, markDeleted bool, mapToAdd map[string]bool) {
+	secondCollectionTemp := make(map[string]interface{})
```

Comment:
Instead of `map[string]interface{}` this should be `map[string]struct{}` and `secondCollectionTemp[secondColUuid]` should be initialized with `struct{}{}`.